test/tests/misc/signal_handler/tests

```diff
@@ -0,0 +1,16 @@
+[Tests]
+	issues = '#12722'
+	[test_signal]
+		type = SignalTester
+		input = 'simple_transient_diffusion.i'
+		requirement = 'The app should write out a checkpoint file at any time by sending a signal to it.'
+
+	[]
+	[test_signal_recover]
+		type = RunApp
+		input = 'simple_transient_diffusion.i'
+		cli_args = '--recover'
+		prereq = 'test_signal'
+		requirement = 'The app should be able to recover from the autosaved checkpoint created by a signal.'
+	[]
+[]
```

Comment:
Add a new line here to get rid of this little guy.

Reply:
More ideas for testing:
- You might want to verify that you can really turn off the outputting of the Checkpoints once they've been turned on. Right now, you won't catch that regression.
- By using RunApp, you aren't verifying that your solution matters. The problem though with checking in a solution file is it's gonna be really big for this input file and we don't really want that. Instead, we may need to do something smarter (like add a few Postprocessors or SideSampler objects) to get the characteristics of our solution without having to look at hundreds of thousands of degrees of freedom for a simple test.

- By not killing the first test, it's possible that your second test does nothing (starts from a finished simulation) and still give a positive result. There's a lot of ways for these two tests to miss the thing you are testing for. You really need to verify that files exist after the signal is sent, and that those files are really being read by your subsequent simulation and that you get the expected result. Anything short of that leaves you open to regressions.